c2rust-analyze/tests/filecheck/type_annotation_rewrite.rs

```diff
@@ -0,0 +1,37 @@
+// CHECK-DAG: pub unsafe fn let_decl()
+pub unsafe fn let_decl() {
+    let mut z = 2;
+    // CHECK-DAG: let x: &(i32);
+    let x: *mut i32;
+    x = std::ptr::addr_of_mut!(z);
+}
+
+// CHECK-DAG: pub unsafe fn let_decl_cell()
+pub unsafe fn let_decl_cell() {
+    let mut z = 2;
+    // CHECK-DAG: let x: &core::cell::Cell<(i32)>;
+    let x: *mut i32;
+    x = std::ptr::addr_of_mut!(z);
```

Comment:
I think there are three issues here:
1. `x = std::ptr::addr_of_mut!(z)` should be rewritten to `x = &z` or `x = &mut z`, depending on the type of `x`
2. `z`'s initializer should be rewritten to `let z = Cell::new(2);` so that `let x: &Cell<i32> = &z;` will work (`z` should have `CELL` in its `addr_of` flags, which is how we know to do this)
3. `z` should be `UNIQUE` and so shouldn't require `Cell` at all (the `*x = *x` operation below should not trigger a borrowck error)

Reply:
Are these things that you'd support having a separate PR for or do you want them addressed here?

Reply:
As per our discussion offline, only rewriting borrows in cases of `Use(Copy)`: f1989754

Reply:
Go ahead.  The `mir_op` part of the implementation should be straightforward.  The `hir_op` side might be a bit tricky, though, since there's a macro involved.